connectors/cli.py

```diff
@@ -140,6 +140,7 @@ def run(args):
     config = {}
     try:
         config = load_config(args.config_file)
+        set_framework_config(config)
```

Comment:
I would avoid this at all costs - having globally exposed config will provoke things that we're heavily trying to avoid - such as connector having access to host configuration, which in turn can lead to config files being used for configuration of individual connectors.

Would it make more sense to do something like this:

1. Make ExtractionService have a class variable `configuration` that will store extraction service config. It's not initialized by default and if ExtractionService is created when no config is provided, then it won't work.

2. Add a class method to update the configuration and use it here, e.g.:
```
class ExtractionService:
    configuration = None
    
    @classmethod
    def update_configuration(class, config):
         class.configuration = config # do validity checks here
    
    def __init__(self):
        self.instance_configuration = type(self).configuration # we treat class as a factory

```

This way config does not leak outside globally + we have a working solution that can easily be changed to anything else.

Reply:
Why not simply use param `config` in `__init__`?

Reply:
>  having globally exposed config will provoke things that we're heavily trying to avoid - such as connector having access to host configuration, which in turn can lead to config files being used for configuration of individual connectors

Why are we trying to avoid that at all costs? We just talked in a sync yesterday about how some individual connector configs should be configurable through the YAML file. I know there was some disagreement on what configs would be a good fit, but I thought there was alignment that this should be possible?

> Make ExtractionService have a class variable configuration that will store extraction service config

If I understand this correctly, the suggestion is still to use a global variable, but to just put it in a place where it's less obviously useful to other code elsewhere? I do not understand the motivation for making it hard for any given code to access our configuration values.

Reply:
> Why not simply use param config in __init__?

I looked at that. But then I'd have to change a ton of method signatures to pass `config` (not just `es_config`) all the way down to whatever is initializing ExtractionService (in this case, `SharepointOnlineDataSource`). 

I'd prefer to be able to look up my configs instead of having to pass them around.